lib/happo/uploader.rb

```diff
@@ -25,26 +25,26 @@ def upload_diffs
             end
 
       diff_images = result_summary[:diff_examples].map do |diff|
-        image = bucket.objects.build(
-          "#{dir}/#{diff[:description]}_#{diff[:viewport]}.png")
+        img_url = "#{dir}/#{diff[:description]}_#{diff[:viewport]}.png"
+        image = bucket.objects.build(img_url)
         image.content = open(Happo::Utils.path_to(diff[:description],
                                                      diff[:viewport],
                                                      'diff.png'))
         image.content_type = 'image/png'
         image.save
-        diff[:url] = image.url
+        diff[:url] = img_url
         diff
       end
 
       new_images = result_summary[:new_examples].map do |example|
-        image = bucket.objects.build(
-          "#{dir}/#{example[:description]}_#{example[:viewport]}.png")
+        img_url = "#{dir}/#{example[:description]}_#{example[:viewport]}.png"
```

Comment:
I should have noticed this before, but these URLs won't work correctly, since they begin with the directory name and not `/`. I think we can shorten these to be just the filename portion for the page, so they are relative to the page.